scipy/stats/_stats_py.py

```diff
@@ -1079,11 +1079,12 @@ def moment(a, order=1, axis=0, nan_policy='propagate', *, center=None):
         calculate_mean = center is None and xp.any(order > 1)
         mean = xp.mean(a, axis=axis, keepdims=True) if calculate_mean else None
         mmnt = []
-        for i in order:
```

Comment:
I imagine you prefer the more explicit variable name, so I'll leave it, but to reduce the diff:
```python3
        for i in (order[j] for j in range(order.shape[0])):
```